core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -196,7 +204,7 @@ void processSubchannelState(Subchannel subchannel, ConnectivityStateInfo stateIn
     if (subchannelData == null || subchannelData.getSubchannel() != subchannel) {
       return;
     }
-    if (newState == SHUTDOWN) {
+    if (newState == SHUTDOWN || rawConnectivityState == SHUTDOWN) {
```

Comment:
if `rawConnectivityState == SHUTDOWN` then `subchannelData == null`. Thus is this change is guaranteed to do nothing. It seems fair to rely on `subchannelData == null`.

Reply:
done.